src/components/logIn/LogIn.vue

```diff
@@ -0,0 +1,180 @@
+<template>
+  <div class="form wrapper">
+    <v-sheet color="white" elevation="5" rounded>
+      <v-form ref="form" v-model="valid" @submit.prevent="submit">
+        <v-text-field
+          v-model="email"
+          :counter="50"
+          :rules="emailRules"
+          :error-messages="errors.email"
+          prepend-icon="mdi-email"
+          label="E-mail"
+          background-color="deep-purple lighten-4"
+          required
+          solo
+          clearable
+        ></v-text-field>
+
+        <v-text-field
+          v-model="password"
+          :counter="50"
+          :rules="passwordRules"
+          :error-messages="errors.password"
+          prepend-icon="mdi-lock"
+          type="password"
+          label="Password"
+          color="deep-purple lighten-4"
+          background-color="deep-purple lighten-4"
+          required
+          solo
+          clearable
+        ></v-text-field>
+        <div class="button wrapper">
+          <v-btn
+            @click="validate"
+            :disabled="!valid"
+            class="mr-4"
+            type="submit"
+            color="success"
+          >
+            submit
+          </v-btn>
+          <v-btn @click="redirectToWelcomePage" color="deep-purple lighten-5">
+            back
+          </v-btn>
+        </div>
+      </v-form>
+    </v-sheet>
+    <v-snackbar
+      v-model="snackbar"
+      vertical="vertical"
+      transition="fab-transition"
+      :timeout="snackbarTimeout"
+      :color="snackbarColor"
+      top
+      right
+      shaped
+      max-width="500"
+      min-width="350"
+      max-height="150"
+      min-height="75"
+    >
+      {{ snackbarMessage }}
+
+      <template v-slot:action="{ attrs }">
+        <v-btn
+          color="brown darken-4"
+          text
+          v-bind="attrs"
+          @click="snackbar = false"
+        >
+          Close
+        </v-btn>
+      </template>
+    </v-snackbar>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import router from '@/router';
+
+export default Vue.extend({
+  name: 'LogIn',
+  data: () => ({
+    valid: false,
+    snackbar: false,
+    snackbarMessage: '',
+    snackbarColor: 'info',
+    snackbarTimeout: 3000,
+    email: '',
+    password: '',
+    errors: { email: '', password: '' },
+    emailRules: [
+      (email: string) =>
+        email?.length >= 10 || 'E-mail cannot be shorter than 10 characters!',
```

Comment:
make sense to move these static strings to a separate file

Reply:
Done